src/update-function/move-array-control.ts

```diff
@@ -0,0 +1,24 @@
+import { MoveArrayControlAction } from '../actions';
+import { formArrayReducer } from '../array/reducer';
+import { FormArrayState } from '../state';
+import { ensureState } from './util';
+
+/**
+ * This update function takes a source index, a destination index, and returns a projection function
+ * that move the child controls at the source index to the destination index from a form array state.
+ */
+export function moveArrayControl<TValue>(from: number, to: number): (state: FormArrayState<TValue>) => FormArrayState<TValue>;
```

Comment:
Please move the generic `TValue` parameter to the returned lambda. This improves the type inference experience.

Also rename parameters to `fromIndex` and `toIndex`.